core/src/components/alert/alert.tsx

```diff
@@ -404,23 +406,22 @@ export class Alert implements ComponentInterface, OverlayInterface {
             return (
               <div class="alert-input-wrapper">
                 <textarea
+                  {...i.attributes as AlertTextareaAttributes}
                   placeholder={i.placeholder}
                   value={i.value}
                   onInput={e => i.value = (e.target as any).value}
                   id={i.id}
                   disabled={i.disabled}
                   tabIndex={0}
-                  class={{
-                    'alert-input': true,
-                    'alert-input-disabled': i.disabled || false
-                  }}
+                  class={inputClass(i)}
                 />
               </div>
             );
           } else {
             return (
               <div class="alert-input-wrapper">
                 <input
+                  {...i.attributes as AlertInputAttributes}
```

Comment:
Let's move this line below `class={inputClass(i)}`. This will allow users to override attributes such as `value` and `id` either in the top level `AlertInput`, or in `attributes`.

For example:

```javascript
{
  attributes: {
    value: 'Hello'
  }
}
```

The above will not do anything since `value={i.value}` is set after `{...i.attributes as AlertInputAttributes}`.

Reply:
Also be sure to do this for the textarea on as well: https://github.com/ionic-team/ionic/pull/21365/files#diff-14f90b10a83b2339d7a4078841868b3eR409

Reply:
I made the change. Later while I was manually testing I had some issues with 3 attributes; onInput, class and disabled. I merged them all for an optimum solution, please and comment on the end result.

Reply:
By the way, changes are all in seperate commits, I can revert easily based on your decision